### DISCUSSION TITLE
Changing model paramaters

### DISCUSSION THREAD

**Discussion by TJABB** [mjhoptics/ray-optics#146] [Q&A] [answered]
Hello,

First, thank you so much for creating Ray-optics! I am currently using it in a project where I have to read a CodeV .seq file into python and modify some parameters of the surfaces. For instance, I would like to change the curvature, thickness, glass and asphere coefficients of a particular surface in the optical model and then update the optical model.  I am currently using the following:

opm = open_model(current_working_directory+"\\dblgauss.seq")
sm = opm['seq_model']
osp = opm['optical_spec']
pm = opm['parax_model']
em = opm['ele_model']
pt = opm['part_tree']

For updating the curvature I am using:
sm.ifcs[1].profile = Spherical(c=1/86)

For updating the thickness I am using
sm.gaps[1].thi = 10

For updating glass:
SFPL51 = create_glass('S-FPL51', 'Ohara')
sm.gaps[1].medium = SFPL51

I haven't discovered how to update the asphere coefficients so far.

May I kindly ask you please to tell me if what I did is the right way of changing the parameters?  It is working but I am not sure if this is the intended way.  Also it would be great if you can tell me how to update the asphere coefficeints?

Many thanks in advance and sorry if this is too basic.

Best Regards,
Toufic

**Comment by TJABB** [comment, marked as answer]
Hello,

Actually I was going through the different Q&As and I found the answer to my question.  Below is  a copy of what I was looking for.

sm.gaps[5].medium = create_glass('N-SK16', 'Schott')
sm.gaps[5].thi = 1.2
sm.ifcs[7].profile.cv = -.12
sm.ifcs[sm.cur_surface].profile = EvenPolynomial(r=63.00, cc=0,
	coefs=[0.0,-4.662529E-6,8.0842E-09,0.0,0.0,0.0])
opm.update_model()
opm.rebuild_from_seq()

**Reply by mjhoptics** [reply]
@TJABB:
Good solution!
Also, if you want to change, say the 4th order coefficient of the polynomial, you can do:
```
poly = sm.ifcs[sm.cur_surface].profile
poly.coefs[1] = -4.5E-6
opm.update_model()
```
In general, you can change the attributes of the various objects directly, including changing multiple different attributes, as you did above, as long as you do an `update_model()` after entering all of the changes.
Using the `rebuild_from_seq()` method is optional for changes that don't change the number of surfaces in the model.  I hope to remove the need for this entirely in the next release.
Regards,
Mike
